Cleanup CI scripts "&&" usage and minor other style changes

SkySkimmer opened the pull request:
Also insert "set -e" in each script, this is redundant with the
ci-common but will behave better if the script is called from the
wrong directory (ie if sourcing ci-common fails).

JasonGross commented:
I'm curious about the reasoning for the change from `&&` to newlines and the other style changes

SkySkimmer commented:
newlines are easier to read and edit
compare
~~~diff
-( cd "${CI_BUILD_DIR}/bedrock2" && git submodule update --init --recursive && COQMF_ARGS='-arg "-async-proofs-tac-j 1"' make && make install )
+( cd "${CI_BUILD_DIR}/bedrock2"
+  git submodule update --init --recursive
+  COQMF_ARGS='-arg "-async-proofs-tac-j 1"' make
+  make install
+)
~~~
For consistency I changed even the shorter ones which just did `cd && make`.

I left the subshell to bound cd effects, this is not really useful since most of the time there is only one subshell at the end of the script and the rest of the time we use `cd $CI_BUILD_DIR/...` which is an absolute path in the next command (the only exception being the lf script IIRC).
OTOH it costs little to I did leave it.

I changed git_download to that it preserves the current directory as I found its behaviour confusing.

I cleaned up some overly verbose variable names in the fiat-crypto scripts and turned them into arrays instead of using word splitting, this is cleaner IMO

If I forgot a change you're curious about ask about it.

ejgallego commented:
Is `set -e` preserved in the subshell ? [I tried to google but was actually not immediate]

JasonGross commented on dev/ci/ci-fiat_crypto.sh:
```diff
@@ -9,16 +11,17 @@ git_download fiat_crypto
+( cd "${CI_BUILD_DIR}/fiat_crypto"
+  git submodule update --init --recursive
+  ulimit -s $stacksize
+  make "${make_args[@]}" pre-standalone-extracted printlite lite
+  make -j 1 "${make_args[@]}" all-except-compiled
```
I guess you leave the targets inline here while factoring them out into arrays in the legacy target because the names are shorter here?

SkySkimmer replied:
yes

SkySkimmer commented:
>Is set -e preserved in the subshell ? [I tried to google but was actually not immediate]

yes
you can test with a script like
~~~bash
set -e 
( echo a; false; echo b )
~~~
which will echo only `a` and exit with code `1`

ejgallego commented:
> you can test with a script like

thanks

ejgallego reviewed (approved):
I think I picked the `&&` style + subshell originally as indeed, while harder to read, makes easier to do cut & paste from the script in the local shell; now this doesn't work so easily, but I Gaëtan prefers this, I'm OK.

Note that you cannot just cut and paste the commands now, but need to go one by one, as you cannot use `set -e` in the main shell either as a failure will kill it, and cut and paste of the whole sequence will run things that shouldn't be run.

So I dunno, still not so sure about that change, but as I wrote, if Gaëtan prefers it after reading this I'm OK.

Rest of changes seem fine.

SkySkimmer commented:
I don't think I've used that copy-paste workflow so I'm not especially concerned about it.

ejgallego commented:
I asked the person from which I learnt this `( cd foo && cmd1 && cmd2 )` and indeed they told me that going `set -e` + a list of commands is considered bad practice in their org, as for example error reporting is harder if you want to tell the user "I failed, here is the shell command you can use to reproduce" , so my soft pref is to keep that style.

ejgallego commented:
> I don't think I've used that copy-paste workflow so I'm not especially concerned about it.

It mostly comes when having to debug stuff.

SkySkimmer commented:
>if you want to tell the user "I failed, here is the shell command you can use to reproduce"

We don't do that so it's irrelevant.

ejgallego commented:
> We don't do that so it's irrelevant

I do it when debugging, so it is. I'm convinced the new style is more fragile.

@coqbot: merge now
